python/ray/data/read_api.py

```diff
@@ -698,6 +704,110 @@ def read_parquet(
     )
 
 
+@PublicAPI
+def read_delta(
+    uri: str,
+    *,
+    filesystem: Optional["pyarrow.fs.FileSystem"] = None,
+    columns: Optional[List[str]] = None,
+    parallelism: int = -1,
+    ray_remote_args: Dict[str, Any] = None,
+    tensor_column_schema: Optional[Dict[str, Tuple[np.dtype, Tuple[int, ...]]]] = None,
+    meta_provider: Optional[ParquetMetadataProvider] = None,
+    **arrow_parquet_args,
+) -> Dataset:
+    """Creates a :class:`~ray.data.Dataset` from delta files.
+
+
+    Examples:
+        Read delta files.
+
+        >>> import ray
+        >>> ds = ray.data.read_delta("/path/to/your/delta/file")
+
+        The Delta reader requires a spark session to be created.
+
+        .. testcode::
+
+            import ray
+            from pyspark.sql import SparkSession
+
+            spark = (SparkSession.builder
+            .config("spark.jars.packages", "io.delta:delta-core_2.12:2.2.0")
```

Comment:
TODO: find a way to get spark scala version and replace the scala version with 2.12 here

Reply:
We can use `spark._jvm.scala.tools.nsc.Properties.versionString()` to get correct scala version.

Reply:
is this for supporting different Scala version? Could we file a TODO issue for it?